proxy/coco/src/request/waiting_room.rs

```diff
@@ -183,14 +181,16 @@ impl<T, D> WaitingRoom<T, D> {
     {
         match self.get(urn) {
             None => Err(Error::MissingUrn(urn.clone())),
-            Some(request) => {
-                match sequence_result(request.clone().transition(matcher, transition))? {
-                    Either::Right(next) => {
-                        self.requests.insert(urn.id.clone(), next.into());
-                        Ok(())
-                    },
-                    Either::Left(mismatch) => Err(Error::StateMismatch((&mismatch).into())),
-                }
+            Some(request) => match request.clone().transition(matcher, transition) {
+                Either::Right(Either::Right(next)) => {
+                    self.requests.insert(urn.id.clone(), next.into());
+                    Ok(())
+                },
+                Either::Right(Either::Left(timeout)) => {
+                    self.requests.insert(urn.id.clone(), timeout.clone().into());
```

Comment:
Does this list every shrink?

Reply:
Not yet: https://github.com/radicle-dev/radicle-upstream/issues/956